x-pack/plugins/cloud_security_posture/public/pages/findings/layout/findings_layout.tsx

```diff
@@ -65,86 +72,164 @@ export const getExpandColumn = <T extends unknown>({
+            'xpack.csp.findings.findingsTable.findingsTableColumn.resourceIdColumnTooltipLabel',
+            {
+              defaultMessage: 'Custom Elastic Resource ID',
+            }
+          )}
+        />
+      ),
+      truncateText: true,
+      width: '15%',
+      sortable: true,
+      render: (filename: string) => (
+        <EuiToolTip position="top" content={filename}>
+          <span>{filename}</span>
+        </EuiToolTip>
+      ),
+    },
+    {
+      field: 'result.evaluation',
+      name: TEXT.RESULT,
+      width: '120px',
+      sortable: true,
+      render: (type: PropsOf<typeof CspEvaluationBadge>['type']) => (
+        <CspEvaluationBadge type={type} />
+      ),
+    },
+    {
+      field: 'resource.sub_type',
+      name: TEXT.RESOURCE_TYPE,
+      sortable: true,
+      width: '150px',
+    },
+    {
+      field: 'resource.name',
+      name: TEXT.RESOURCE_NAME,
+      sortable: true,
+    },
+    {
+      field: 'rule.name',
+      name: TEXT.RULE,
+      sortable: true,
+    },
+    {
+      field: 'rule.section',
+      name: TEXT.CIS_SECTION,
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'cluster_id',
+      name: (
+        <ColumnNameWithTooltip
+          columnName={TEXT.CLUSTER_ID}
+          tooltipContent={i18n.translate(
+            'xpack.csp.findings.resourceTable.resourceTableColumn.clusterIdColumnTooltipLabel',
+            {
+              defaultMessage: 'Kube-System Namespace ID',
+            }
+          )}
+        />
+      ),
+      truncateText: true,
+      sortable: true,
+    },
+    {
+      field: '@timestamp',
+      width: '150px',
+      name: TEXT.LAST_CHECKED,
+      truncateText: true,
+      sortable: true,
+      render: (timestamp: number) => (
+        <EuiToolTip position="top" content={timestamp}>
+          <span>{moment(timestamp).fromNow()}</span>
+        </EuiToolTip>
+      ),
+    },
+  ].map((column) => (onAddFilter ? createColumnWithFilters(column, { onAddFilter }) : column));
+
+export const createColumnWithFilters = <T extends unknown>(
+  column: EuiTableFieldDataColumnType<T>,
+  { onAddFilter }: { onAddFilter(key: string, value: Serializable, negate: boolean): void }
+): EuiTableFieldDataColumnType<T> => ({
+  ...column,
+  render: (cellValue: Serializable, item: T) => (
+    <FilterableCell
+      onAddFilter={() => onAddFilter(column.field as string, cellValue, false)}
+      onAddNegateFilter={() => onAddFilter(column.field as string, cellValue, true)}
+      field={column.field as string}
+    >
+      {column.render?.(cellValue, item) || getCellValue(cellValue)}
+    </FilterableCell>
+  ),
+});
+
+const getCellValue = (value: unknown) => {
+  if (!value) return;
+  if (typeof value === 'string' || typeof value === 'number') return value;
+};
+
+const FilterableCell: React.FC<{
+  onAddFilter(): void;
+  onAddNegateFilter(): void;
+  field: string;
+}> = ({ children, onAddFilter, onAddNegateFilter, field }) => (
```

Comment:
`FilterableCell` (thanks @JordanSh) is used as a custom solution to integrate cell filters similar to discover plugin, without using the eui data grid. (for now)

Reply:
@tinnytintin10 - followup: this is the filter button implementation 

it's similar to one of discover's impl on the sidebar - 

https://github.dev/elastic/kibana/blob/0d4cc4dc326f1475070862c0fb11414883768ef5/src/plugins/discover/public/application/main/components/sidebar/discover_sidebar.scss#L48-L72

note to self: add more :pseudo states like above^